### PR TITLE
Added controllers for top users, products and words

### PR DESCRIPTION
**Pull request by Nokafan**

### CODE REVIEW

File: src/main/java/spring/boot/parser/controllers/ProductsController.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import spring.boot.parser.dto.ProductResponseDto;
+import spring.boot.parser.mapper.ProductMapper;
+import spring.boot.parser.service.ProductService;
+
+@RestController
+@RequestMapping("/product")
```
Comment (boroda4436) [inline src/main/java/spring/boot/parser/controllers/ProductsController.java]:
```suggestion
@RequestMapping("/products")
```

File: src/main/java/spring/boot/parser/controllers/UsersController.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import spring.boot.parser.dto.UserResponseDto;
+import spring.boot.parser.mapper.UserMapper;
+import spring.boot.parser.service.UserService;
+
+@RestController
+@RequestMapping("/user")
```
Comment (boroda4436) [inline src/main/java/spring/boot/parser/controllers/UsersController.java]:
```suggestion
@RequestMapping("/users")
```

File: src/main/java/spring/boot/parser/controllers/WordsController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import spring.boot.parser.dto.WordResponceDto;
+import spring.boot.parser.mapper.WordsMapper;
+import spring.boot.parser.service.WordService;
+
+@RestController
+@RequestMapping("/word")
```
Comment (boroda4436) [inline src/main/java/spring/boot/parser/controllers/WordsController.java]:
```suggestion
@RequestMapping("/words")
```

File: src/main/java/spring/boot/parser/service/UserService.java [hunk 4]
```diff
@@ -1,7 +1,11 @@
 package spring.boot.parser.service;
 
+import java.util.List;
+import org.springframework.data.domain.Pageable;
 import spring.boot.parser.model.User;
 
 public interface UserService extends GeneralService<User> {
     User getByUserId(String externalUserId);
```
Comment (boroda4436) [inline src/main/java/spring/boot/parser/service/UserService.java]:
```suggestion
    User getByExternalUserId(String externalUserId);
```

**Review by boroda4436 — APPROVED** [review]
LGTM

File: src/main/java/spring/boot/parser/controllers/ProductsController.java (hunk 1 above)
Comment (Nokafan) [reply src/main/java/spring/boot/parser/controllers/ProductsController.java]: Fixed

File: src/main/java/spring/boot/parser/controllers/UsersController.java (hunk 2 above)
Comment (Nokafan) [reply src/main/java/spring/boot/parser/controllers/UsersController.java]: Fixed

File: src/main/java/spring/boot/parser/controllers/WordsController.java (hunk 3 above)
Comment (Nokafan) [reply src/main/java/spring/boot/parser/controllers/WordsController.java]: Fixed

File: src/main/java/spring/boot/parser/service/UserService.java (hunk 4 above)
Comment (Nokafan) [reply src/main/java/spring/boot/parser/service/UserService.java]: Fixed

**Pull request merged**